src/input/source/led/multicolor_chassis.rs

```diff
@@ -0,0 +1,167 @@
+use crate::{
+    config::LedFixedColor,
+    input::source::{SourceInputDevice, SourceOutputDevice},
+    udev::device::UdevDevice,
+};
+use std::{error::Error, fmt::Debug, path::PathBuf};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum MultiColorChassisError {
+    #[error("Error reading multi_index: {0}")]
+    MultiIndexError(std::io::Error),
+
+    #[error("File multi_intensity not found: {0}")]
+    MultiIntensityNotFound(PathBuf),
+
+    #[error("Error updating multi_intensity: {0}")]
+    MultiIntensityUpdateError(std::io::Error),
+
+    #[error("Unsupported index type: {0}")]
+    UnsupportedIndexType(String),
+}
+
+enum IndexType {
+    RGB,
+}
+
+/// MultiColorChassis source device implementation
+pub struct MultiColorChassis {
+    multi_intensity_path: PathBuf,
+    multi_index_map: Vec<IndexType>,
+    fixed_color: Option<LedFixedColor>,
```

Comment:
Do we ever use `fixed_color` outside initialization? Maybe we can remove this and just set the color in the constructor?

Reply:
If we want to set it back after a resume or keep track of it I left it just for good measure, but yeah not very usefult.

Reply:
Are we not storing the last setting otherwise? We could call this current_color and report it in dbus.